tests/auxtel/test_latiss_cwfs_align.py

```diff
@@ -0,0 +1,512 @@
+                self.atheaderservice.close(),
+                self.ataos.close(),
+                self.athexapod.close(),
+                self.atptg.close(),
+                self.atmcs.close(),
+            )
+            logger.debug("Remotes Closed")
+
+    async def cmd_take_images_callback(self, data):
+        logger.debug(f"cmd_take_images callback came with data of {data}")
+        one_exp_time = (
+            data.expTime
+            + self.script.latiss.read_out_time
+            + self.script.latiss.shutter_time
+        )
+        logger.debug(
+            f"Exposing for {one_exp_time} seconds for each exposure, total exposures is {data.numImages}"
+        )
+        await asyncio.sleep(one_exp_time * data.numImages)
+        self.nimages += 1
+        logger.debug("Scheduling finish_take_images before returning from take_images")
+        self.end_image_tasks.append(asyncio.create_task(self.finish_take_images()))
+
+    async def finish_take_images(self):
+        await asyncio.sleep(0.5)
+        imgNum = self.atcamera.cmd_takeImages.callback.await_count - 1
+        image_name = f"AT_O_{self.date}_{(imgNum + self.seq_num_start):06d}"
+        self.atcamera.evt_endReadout.set_put(imageName=image_name)
+        await asyncio.sleep(0.5)
+        self.atheaderservice.evt_largeFileObjectAvailable.put()
+        await asyncio.sleep(1.0)
+        self.atarchiver.evt_imageInOODS.set_put(obsid=image_name)
+
+    async def test_configure(self):
+        async with self.make_script():
+            # Try configure with minimum set of parameters declared
+            grating = "test_disp1"
+            filter = "test_filt1"
+            exposure_time = 1.0
+            await self.configure_script(
+                grating=grating,
+                filter=filter,
+                exposure_time=exposure_time,
+                dataPath=DATAPATH,
+            )
+
+    async def atcs_get_bore_sight_angle(self):
+        """Returns nasmyth rotator value for image"""
+        logger.debug(f"visit_id_angles is: {self.visit_id_angles}")
+        logger.debug(
+            f"extra_visit_id is {self.script.extra_visit_id} and of type {type(self.script.extra_visit_id)}"
+        )
+        if self.script.extra_visit_id not in self.visit_id_angles:
+            raise IOError(
+                f"Image ID of {self.script.extra_visit_id} is not contained"
+                " in self.visit_id_angles"
+            )
+
+        # modified from atcs.py
+        # instrument on nasymth2, so
+        parity_x = -1
+        elevation_angle = self.visit_id_angles[self.script.extra_visit_id][0]
+        nasmyth_angle = self.visit_id_angles[self.script.extra_visit_id][1]
+        bore_sight_angle = elevation_angle + parity_x * nasmyth_angle + 90.0
+
+        return bore_sight_angle
+
+    @unittest.skipIf(
+        DATA_AVAILABLE is False,
+        "Data not available for test_take_sequence, skipping.",
+    )
+    async def test_take_sequence(self):

```

Comment:
We really want to run this on several datasets.. not just one.

Reply:
Although I agree it really makes me wonder, how much of this is testing the script and how much is testing the source selection algorithm. I think we have to separate these two things. I don't think we should expect Scripts to re-test external libraries.

Reply:
Discussed in person, will create a new repo for these types of analyses in the future. This script will remain unchanged since it'll be significantly refactored once the MTAOS is ready.